src/esp/physics/ArticulatedObject.h

```diff
@@ -553,6 +667,102 @@ class ArticulatedObject : public esp::physics::PhysicsObjectBase {
     Magnum::Debug{} << "No base implementation of \"clampJointLimits\". ";
   }
 
+  //=========== Joint Motor API ===========
+
+  /**
+   * @brief Create a new JointMotor from a JointMotorSettings.
+   *
+   * Note: No base implementation. See @ref bullet::BulletArticulatedObject.
+   * @param index DoF (for revolute or prismatic joints) or Link (spherical
+   * joints)
+   * @param settings The settings for the joint motor. Must have JointMotorType
+   * correctly configured.
+   * @return The motorId for the new joint motor or ID_UNDEFINED (-1) if failed.
+   */
+  virtual int createJointMotor(
+      CORRADE_UNUSED const int index,
+      CORRADE_UNUSED const JointMotorSettings& settings) {
+    Magnum::Debug{} << "No base implementation of \"createJointMotor\". "
+                       "Requires a physics simulator implementation.";
+    return ID_UNDEFINED;
+  }
+
+  /**
+   * @brief Remove and destroy a joint motor.
+   */
+  virtual void removeJointMotor(const int motorId) {
+    CHECK(jointMotors_.count(motorId) > 0);
+    jointMotors_.erase(motorId);
+  }
+
+  /**
+   * @brief Get a copy of the JointMotorSettings for an existing motor.
+   */
+  virtual JointMotorSettings getJointMotorSettings(const int motorId) {
+    CHECK(jointMotors_.count(motorId) > 0);
```

Comment:
`CHECK` is fatal from the glog:

```
// CHECK dies with a fatal error if condition is not true.  It is *not*
// controlled by DCHECK_IS_ON(), so the check will be executed regardless of
// compilation mode.  Therefore, it is safe to do things like:
//    CHECK(fp->Write(x) == 4)
#define CHECK(condition)  \
      LOG_IF(FATAL, GOOGLE_PREDICT_BRANCH_NOT_TAKEN(!(condition))) \
             << "Check failed: " #condition " "
```

A sanity check, adding `CHECK(jointMotors_.count(motorId) == 0);` here produces:

```
F0617 13:17:18.694432 399650304 ArticulatedObject.h:703] Check failed: jointMotors_.count(motorId) == 0 
*** Check failure stack trace: ***
Fatal Python error: Aborted

Current thread 0x0000000117d22e00 (most recent call first):
  File "...habitat-sim/tests/test_physics.py", line 1188 in test_articulated_object_joint_motors
... more stack trace
```

Reply:
The macro override in `esp/core/logging.h`, however may not be fatal but appears to be unused. Perhaps this is not in line with our expectations. Happy to change this to ESP_CHECK. Should probably refactor several places in the code for consistency at some point.

Reply:
Ok, I didn't notice that `CHECK` was fatal. But yes, for a fatal user error, we settled on `ESP_CHECK`. One nice thing is the way it bubbles up errors to python if you're running python (so you get a python callstack instead of just aborting the whole process).

Reply:
See also comment at top of src/esp/core/Check.h

Reply:
>it actually doesn't link the C++ line of the error. 
We could certainly add __FILE__ and __LINE__ inside the ESP_CHECK macro. At the time I implemented ESP_CHECK, I guess I figured any user could search for the error string to find the C++ source. Additionally, in theory a user shouldn't care about the source C++ code since she generally doesn't have access to it.